projects/igniteui-angular/src/lib/grids/grid-public-cell.ts

```diff
@@ -60,6 +60,21 @@ export class IgxGridCell implements CellType {
         return this._column || this.grid.getColumnByName(this._columnField);
     }
 
+    private searchColumnIndexByName(columnIndex: number, name: string): any {
+        const columns = this.grid.getAllColumnsByName(name);
+
+        if (columns.length === 0 || columnIndex < 0 || columnIndex >= columns.length) {
+            return -1;
+        }
+    
+        const columnAtIndex = columns[columnIndex];
+        if (columnAtIndex.field === name) {
```

Comment:
the columnAtIndex is initally retreieved by filtering columns by name (line 64). Then this columnn is compared again against the same `name`).